src/semaphore_internal.h

```diff
@@ -29,6 +29,13 @@
 
 struct dispatch_queue_s;
 
+#if USE_FUTEX_SEM
+typedef struct {
+	/* Low 32 bits are the count; high bits are the waiter count. */
+	volatile uint64_t dfx_data;
+} dispatch_futex_t;
+#endif  /* USE_FUTEX_SEM */
```

Comment:
reload what?? of course you don't have to, you can use the value through the unt64_t, and pass the &dfx->dfx_futex to the futex call alright. it's just an addres, there's nothing to read from it, only the kernel does.